tests/e2e/slack_test.go

```diff
@@ -0,0 +1,366 @@
+//go:build integration
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/slack-go/slack"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/vrischmann/envconfig"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/infracloudio/botkube/pkg/filterengine/filters"
+)
+
+type Config struct {
+	KubeconfigPath string `envconfig:"optional,KUBECONFIG"`
+	Deployment     struct {
+		Name          string        `envconfig:"default=botkube"`
+		Namespace     string        `envconfig:"default=botkube"`
+		ContainerName string        `envconfig:"default=botkube"`
+		WaitTimeout   time.Duration `envconfig:"default=3m"`
+		Envs          struct {
+			SlackEnabledName   string `envconfig:"default=COMMUNICATIONS_SLACK_ENABLED"`
+			SlackChannelIDName string `envconfig:"default=COMMUNICATIONS_SLACK_CHANNEL"`
+		}
+	}
+	ClusterName string `envconfig:"default=sample"`
+	Slack       SlackConfig
+}
+
+type SlackConfig struct {
+	BotName                  string `envconfig:"default=botkube"`
+	AdditionalContextMessage string `envconfig:"optional"`
+	TesterAppToken           string
+	MessageWaitTimeout       time.Duration `envconfig:"default=10s"`
+}
+
+const (
+	channelNamePrefix = "test"
+	welcomeText       = "Let the tests begin 🤞"
+	pollInterval      = time.Second
+)
+
+func TestSlack(t *testing.T) {
```

Comment:
Initially I wanted to go with `TestMain` approach to prepare test setup, but the code would be a bit more complicated (error handling) and also I'm not really sure if that would be valid. If we added new tests e.g. for Discord bot etc., we would extract the test setup part for Slack to different location anyway.

So I went with a simple approach (IMHO the most readable one) of running subtests, but I'm open to other approaches (creating test suite etc.).